src/socket/recv.rs

```diff
@@ -143,14 +143,26 @@ impl RecvHandler {
             return;
         }
         // Decodes the packet
-        let (packet, authenticated_data) =
-            match Packet::decode::<P>(&self.node_id, &self.recv_buffer[..length]) {
-                Ok(p) => p,
-                Err(e) => {
-                    debug!("Packet decoding failed: {:?}", e); // could not decode the packet, drop it
-                    return;
+        let (packet, authenticated_data) = match Packet::decode::<P>(
+            &self.node_id,
+            &self.recv_buffer[..length],
+        ) {
+            Ok(p) => p,
+            Err(e) => {
+                // This could be a packet to keep a NAT hole punched for this node in the
+                // sender's NAT, hence only serves purpose for the sender.
+                if length == 0 {
+                    debug!("Appears to be a packet to keep a hole punched in sender's NAT, dropping. src: {}", src_address);
```

Comment:
```suggestion
                    debug!("Received packet requesting to keep sender's NAT hole punched, dropping. src: {}", src_address);
```
The debug was a little confusing, maybe this makes it more clear?

Reply:
refer to comment above, nothing is being requested, the sent packet only has purpose for the sender but will arrive at the receiver anyhow